tarsier/core.py

```diff
@@ -1,20 +1,43 @@
 from asyncio import Protocol
 from pathlib import Path
-from typing import Dict, Tuple
+from typing import Tuple, Optional
 
 from tarsier._utils import load_js
 from tarsier.adapter import AnyDriver, BrowserAdapter, adapter_factory
 from tarsier.ocr import OCRService
 from tarsier.text_format import format_text
 
-TagToXPath = Dict[int, str]
+
+class TagMetadata:
+    def __init__(
+        self,
+        tarsierID: int,
+        elementName: str,
+        elementHTML: str,
+        xpath: str,
+        elementText: Optional[str],
+        textNodeIndex: Optional[int],
+        idSymbol: str,
+        idString: str,
+    ):
+        self.tarsierID = tarsierID
+        self.elementName = elementName
+        self.elementHTML = elementHTML
+        self.xpath = xpath
+        self.elementText = elementText
+        self.textNodeIndex = textNodeIndex
+        self.idSymbol = idSymbol
+        self.idString = idString
+
+    def __repr__(self) -> str:
+        return f"TagMetadata(tarsierID={self.tarsierID}, elementName={self.elementName}, xpath={self.xpath})"
```

Comment:
We should use TypedDict's to represent data classes instead of a straight up class